webpack.config.js

```diff
@@ -0,0 +1,169 @@
+import path from 'path';
+import { fileURLToPath } from 'url';
+import ReactRefreshPlugin from '@pmmmwh/react-refresh-webpack-plugin';
+import plugin from 'eslint-plugin-react';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+// Сначала определяем __filename и __dirname
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const srcPath = path.resolve(__dirname, 'src');
+import Dotenv from 'dotenv-webpack';
+import TsCheckerPlugin from 'fork-ts-checker-webpack-plugin';
+// Теперь можно использовать __dirname для определения buildPath
+const buildPath = path.resolve(__dirname, 'dist');
+
+const isProd = process.env.NODE_ENV === 'production';
+const getSettingsForStyles = (withModules = false) => {
+  // Заменяем в нашей функции style-loader на mini-css-extract-plugin
+  return [
+    isProd ? MiniCssExtractPlugin.loader : 'style-loader',
+    !withModules
+      ? 'css-loader'
+      : {
+          loader: 'css-loader',
+          options: {
+            modules: {
+              localIdentName: !isProd ? '[path][name]__[local]' : '[hash:base64]',
```

Comment:
[+] сейчас имена у стилей получаются слишком длинными, особенно с увеличением вложенности. возможно, будет читабельнее вместо пути добавлять небольшой хэш:
```suggestion
              localIdentName: !isProd ? '[name]__[local]__[hash:base64:5]' : '[hash:base64]',
```